diff --git a/node_modules/react-native-webview/apple/RNCWebViewImpl.m b/node_modules/react-native-webview/apple/RNCWebViewImpl.m
index 97d9780..c05b394 100644
--- a/node_modules/react-native-webview/apple/RNCWebViewImpl.m
+++ b/node_modules/react-native-webview/apple/RNCWebViewImpl.m
@@ -507,7 +507,7 @@ - (WKWebViewConfiguration *)setUpWkWebViewConfig
 - (void)initializeWebView
 {
   @synchronized (self) {
-    if (_webView != nil) {
+    if (self.superview == nil || _webView != nil) {
       return;
     }
     WKWebViewConfiguration *wkWebViewConfig = [self setUpWkWebViewConfig];
@@ -564,21 +564,9 @@ - (void)initializeWebView
   }
 }
 
-// react-native-mac os does not support didMoveToSuperView
-#if !TARGET_OS_OSX
-- (void)didMoveToSuperview
-{
-  if (self.superview != nil && _webView == nil) {
-    [self initializeWebView];
-  }
-}
-#endif // !TARGET_OS_OSX
-
 - (void)didMoveToWindow
 {
-  if (self.window != nil && _webView == nil) {
-    [self initializeWebView];
-  }
+  [self initializeWebView];
 
 #if !TARGET_OS_OSX
   // Allow this object to recognize gestures
@@ -798,6 +786,8 @@ - (void)setSource:(NSDictionary *)source
   if (![_source isEqualToDictionary:source]) {
     _source = [source copy];
 
+    [self initializeWebView];
+
     if (_webView != nil) {
       [self visitSource];
     }
